floracast/steps/evaluate.py

```diff
@@ -0,0 +1,352 @@
+                img {{ max-width: 100%; height: auto; }}
+            </style>
+        </head>
+        <body>
+            <div class="header">
+                <h1>FloraCast Model Evaluation</h1>
+            </div>
+            
+            <div class="metrics">
+                <h2>Evaluation Metrics</h2>
+                <p><strong>Metric:</strong> {metric.upper()}</p>
+                <p><strong>Score:</strong> {score:.4f}</p>
+                <p><strong>Prediction Horizon:</strong> {len(predictions)} time steps</p>
+                <p><strong>Training Data Points:</strong> {len(train_series)}</p>
+                <p><strong>Validation Data Points:</strong> {len(val_series)}</p>
+            </div>
+            
+            <div class="plot">
+                <h2>Time Series Visualization</h2>
+                <img src="data:image/png;base64,{img_base64}" alt="Evaluation Plot">
+            </div>
+            
+            <div class="footer">
+                <p><em>Generated by FloraCast evaluation step</em></p>
+            </div>
+        </body>
+        </html>
+        """
+
+        return HTMLString(html_content)
+
+    except Exception as e:
+        logger.error(f"Failed to create visualization: {str(e)}")
+        # Return simple HTML with error message
+        error_html = f"""
+        <html>
+        <body>
+            <h1>Evaluation Results</h1>
+            <p><strong>Metric:</strong> {metric.upper()}</p>
+            <p><strong>Score:</strong> {score:.4f}</p>
+            <p><em>Visualization failed to generate: {str(e)}</em></p>
+        </body>
+        </html>
+        """
+        return HTMLString(error_html)
+
+
+@step
+def evaluate(
+    model: object,
+    train_series: TimeSeries,
+    val_series: TimeSeries,
+    horizon: int = 7,
+    metric: str = "smape",
+) -> Tuple[
+    Annotated[float, "evaluation_score"],
+    Annotated[HTMLString, "evaluation_visualization"],
+]:
+    """
+    Evaluate the trained model on validation data.
+
+    Args:
+        model: Trained forecasting model
+        train_series: Training time series
+        val_series: Validation time series
+        horizon: Forecasting horizon
+        metric: Evaluation metric name
+
+    Returns:
+        Evaluation metric score (lower is better for SMAPE)
+    """
+
+    logger.info(f"Evaluating with horizon: {horizon}, metric: {metric}")
+
+    try:
+        # Generate predictions using TFT model
+        # TFT requires the series parameter to generate predictions
+        logger.info(f"Generating predictions for horizon: {horizon}")
+
+        # For TFT models, we need to provide the series parameter
+        if hasattr(model, "predict"):
+            # Generate predictions using iterative multi-step approach for longer horizons
+            # This is better than single-shot prediction for long horizons
+            n_predict = min(
+                len(val_series), 42
+            )  # Cap at 6 weeks for reasonable evaluation
```

Comment:
This never uses the horizon param from the config. So you can't control the evaluation horizon from YAML even though it's exposed. Probably cap at `len(val_series)` instead.

Reply:
Got rid of ExponentialSmoothing was just adding complexity